### PR TITLE
Update workflow for yast2 sw_single module

### PR DESCRIPTION
**Pull request by mloviska**
- Related ticket: [[functional][y] test fails in yast2_i - package has preselected, seems a new feature from yast2-packager 4.1.27?](https://progress.opensuse.org/issues/48125)
- Needles: [o3 - tw only so far](https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/516)
- Verification runs: 
   * [tw-gnome](http://eris.suse.cz/tests/12096#step/yast2_i/1)
   * [leap15.1](http://eris.suse.cz/tests/12108#step/yast2_i/1)
   * [sle15sp1](http://eris.suse.cz/tests/12107#step/yast2_i/1)
   * [sle12sp5](http://eris.suse.cz/tests/12106#step/yast2_i/1)
   * [leap15.0](http://eris.suse.cz/tests/12109#step/yast2_i/1)

### CODE REVIEW

File: tests/console/yast2_i.pm [hunk 1]
```diff
@@ -54,6 +54,17 @@ sub run {
         }
         wait_screen_change { send_key 'alt-p' };
     }
+
+    # in new yast2 sw_install we need to change filter to Search
+    if (is_tumbleweed) {
```
Comment (okurz) [inline tests/console/yast2_i.pm]: Our tests should always consider openSUSE Tumbleweed as the default product without a version. That means any differing behaviour should have an explicit exclude rule for the "older products" and potentially an explanation why the behaviour should differ. You can try to negate the logic check.

Comment (rwx788) [reply tests/console/yast2_i.pm]: I believe we'll get this change to SLE 15 SP1 and leap 15.1, what's about using `yast2-preselected-driver` tag for the detection?

**Comment by mloviska** [comment]
@rwx788 atm cloning qt!

File: tests/console/yast2_i.pm (hunk 1 above)
Comment (mloviska) [reply tests/console/yast2_i.pm]: It is not only in the preselect package, but we need to navigate to filter and select search option as well. Search has been the default opt until the newest updates.

Comment (okurz) [reply tests/console/yast2_i.pm]: ok, could you still negate the check and do e.g. `if (!is_leap('<=15.1') && !is_sle('<=15-SP1'))`

Comment (mloviska) [reply tests/console/yast2_i.pm]: I think it is easier and more clear if we include only the bare minimum in the condition. I guess that we can expect this change will hit the newest leap and sle15sp1 anyway.

**Comment by rwx788** [comment]
Sorry, seems I'm missing something, but could you please provide link to the additional needles for qt (if we need any).

**Comment by mloviska** [comment]
@rwx788 yast2_i module runs only in the console/ncurses.

**Comment by rwx788** [comment]
Ok, that's true, but as it's 100% same problem as in https://openqa.opensuse.org/tests/860424#step/yast2_software_management/15 can we apply common solution for both? Otherwise I will create separate task.

**Comment by mloviska** [comment]
Module *yast2_software_management* just opens and closes the management window. It would be enough just to update the needle on o3 side. Personally, I guess one would spend more time with the ticket than resolving the issue.

**Comment by mloviska** [comment]
But maybe we could merge both test modules into one

**Comment by rwx788** [comment]
Makes sense, I will just create needle, as currently needling on o3 is broken.

**Comment by rwx788** [comment]
> But maybe we could merge both test modules into one

Yeah, that would make sense. Recently we did same for yast2_lan as previously we have maintained same test in 2 places which started to differ =) I will file a ticket.

**Comment by rwx788** [comment]
> But maybe we could merge both test modules into one

Yeah, that would make sense. Recently we did same for yast2_lan as previously we have maintained same test in 2 places which started to differ =) I have filed a ticket: https://progress.opensuse.org/issues/48542

**Pull request merged**